schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,257 @@
+>    prioritized order during content-type negotiation. If a service offers up
+>    the same entity in JSON, Protobuf, and Avro formats, there would therefore
+>    be three distinct schemas, each defining a representation of the data
+>    entity for the respective format. From the same "REST" mindset perspective,
+>    the ideal representation of this set of schemas, which differ with regards
+>    to the format they describe, but ought to be semantically identical, is to
+>    also differentiate them only by their content type and maintain them under
+>    the exact same name as if they were one schema.
+> 2) Since the above strategy is truly RESTful, but quite esoteric if you've not
+>    grown up as a RESTafarian, the alternative strategy for concurrently
+>    handling multiple schema formats is much simpler: Constrain each schema
+>    group to a single format.
+
+## 2. Schema Registry Elements
+
+This section further describes the elements enumerated in the introduction.
+
+### 2.1. Schema Group
+
+A schema group is a named collection of schemas. Since a schema registry is
+often a resource with a scope greater than a single application and might even
+span multiple organizations, it is very useful to put a grouping construct
+around sets of schemas that are related either by ownership or by a shared
+subject matter context.
+
+Implementations of this specification MAY associate access control rules with
+schema groups. For instance, a user or a group of users might be given write
+access only to a particular schema group that their organization owns. If trade
+secret protection is required for an application or parts of it and the schema
+structure would give some of those away, read access to a group of schemas might
+likewise be restricted. 
+
+Access control rules at the schema group level MUST NOT limit visibility of the
+schema groups themselves to users authorized to enumerate schema groups at the
+registry level or to create new schema groups in order to create unambiguous
+transparency about which `id` have already been taken. An implementation MAY
+withhold all details other than the `id` attribute from a user not authorized
+to access the schema group.
+
+This specification does not define management constructs for such access control
+rules.
+
+### 2.1.1. Schema group attributes
+
+The data model for a schema group consists of three attributes:
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema group.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax
+  - MUST be unique within the scope of the schema registry
+- Examples:
+  - mygroup
+  - my-group
+
+#### format
+
+- Type: `String`
+- Description: Defines the schema format managed by this schema group. The
+  formats supported by an implementation are specific to the implementation.
+  This specification does not mandate support for particular formats.
+- Constraints:
```

Comment:
Do we want to prescribe that this SHOULD be the media type for the schema as defined by the schema spec? For example, JSONSchema proposes "application/json+schema": https://json-schema.org/draft/2019-09/json-schema-core.html

Additionally, the schema format specs themselves are versioned, I wonder if this is something we need to account for here or whether we should leave that up to implementations.

Reply:
If we differentiate by media type, we would need to backfill definitions that are are not provided by the respective standards; best as vendor extensions such that we don't get in the way of them changing their minds, e.g. "application/vnd.cloudnative.schema-registry.avro"

Reply:
I don't have a strong opinion here - but agree that media types can be problematic for the reason given by @evankanderson  - lots of formats don't have registered mime types. :(  Mime type does benefit from being nicely extensible (vs. something like an enum).  I presume it's not the goal of this spec to mandate the supported schema formats an implementation must support.